File: msg-key-phrases/src/test/java/dev/number6/keyphrases/dagger/TestChannelMessagesKeyPhrasesComprehensionComponent.java
package dev.number6.keyphrases.dagger;

import dev.number6.comprehend.results.PresentableKeyPhrasesResults;
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction;
import dev.number6.message.ComprehensionResultsConsumer;
import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {
        FakeCommonModule.class,
        ChannelMessagesHandlerModule.class,
        FakeComprehensionResultsModule.class
})
@Singleton
public interface TestChannelMessagesKeyPhrasesComprehensionComponent extends ChannelMessagesKeyPhrasesComprehensionComponent {

    @Singleton
    ChannelMessagesToComprehensionResultsFunction<PresentableKeyPhrasesResults> getResultsFunction();

    @Singleton
    ComprehensionResultsConsumer<PresentableKeyPhrasesResults> getConsumer();
}

File: msg-key-phrases/src/test/java/dev/number6/keyphrases/dagger/TestChannelMessagesKeyPhrasesComprehensionComponent.kt
package dev.number6.keyphrases.dagger

import dagger.Component
import dev.number6.comprehend.results.PresentableKeyPhrasesResults
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction
import dev.number6.message.ComprehensionResultsConsumer
import javax.inject.Singleton

@Component(modules = [FakeCommonModule::class, ChannelMessagesHandlerModule::class, FakeComprehensionResultsModule::class])
@Singleton
interface TestChannelMessagesKeyPhrasesComprehensionComponent : ChannelMessagesKeyPhrasesComprehensionComponent {
    @Singleton
    fun getResultsFunction(): ChannelMessagesToComprehensionResultsFunction<PresentableKeyPhrasesResults>

    @Singleton
    fun getConsumer(): ComprehensionResultsConsumer<PresentableKeyPhrasesResults>
}